### PR TITLE
Move json token cache to input

### PR DESCRIPTION
**Pull request by yufeih**
Manage JSON/YAML token cache in `Input`. 

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/docfx/pull/5210)

### CODE REVIEW

File: test/docfx.Test/TestUtility.cs [hunk 1]
```diff
@@ -66,7 +66,7 @@ void Throw(string stackTrace, string message, string detailMessage, string info)
 
         private static void CreateZipFile(KeyValuePair<string, string> file, string filePath)
         {
-            var token = YamlUtility.ToJToken(file.Value);
+            var token = YamlUtility.ToJToken(new StringReader(file.Value));
```
Comment (live1206) [inline test/docfx.Test/TestUtility.cs]: why not put `new StringReader()` into `ToJToken` as well, or another overloading for it?

File: src/docfx/build/context/Cache.cs [hunk 2]
```diff
@@ -4,16 +4,11 @@
...
 using System.Collections.Concurrent;
 using System.Collections.Generic;
-using System.IO;
-using Newtonsoft.Json.Linq;
 
 namespace Microsoft.Docs.Build
 {
     internal class Cache
     {
-        private readonly ConcurrentDictionary<string, Lazy<(List<Error>, JToken)>> _tokenCache
```
Comment (live1206) [inline src/docfx/build/context/Cache.cs]: what are we benefiting by moving the JSON/YAML cache from `Cache.cs` to `Input.cs`?
Or why do we leave TOC cache in `Cache.cs`?

Comment (live1206) [reply src/docfx/build/context/Cache.cs]: Just noticed #5211, so the idea is to get rid of `Cache.cs`

File: test/docfx.Test/TestUtility.cs (hunk 1 above)
Comment (yufeih) [reply test/docfx.Test/TestUtility.cs]: Either is ok, `ToJToken` is more of a private method, I initially extract it to share between docfx and docastest. Didn't thought it is used here. This line can call `Deserialize<JToken>` instead

**Pull request merged**